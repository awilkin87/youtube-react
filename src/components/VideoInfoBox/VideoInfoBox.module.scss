@use '../../styles/shared.module';

.container {
  $channel-image-size: calc(#{shared.$avatar-diameter + shared.$avatar-margin});

  display: grid;
  grid-template: "channel-image video-info  subscribe"   auto
                 "blank-left    description blank-right" auto
                 /max-content     auto        max-content;
  align-items: center;
  row-gap: 16px;
  color: shared.$primary-text-color;

  .channelImage {
    grid-area: channel-image;
    @include shared.left-avatar;
  }

  .videoInfo {
    grid-area: video-info;

    .channelName {
      font-weight: 600;
    }

    .subscriberCount {
      color: shared.$secondary-text-color;
      font-size: 13px;
    }
  }

  .subscribeButton {
    grid-area: subscribe;
    border-radius: 0; // Override SemanicUI border
    text-transform: uppercase;
  }

  .description {
    grid-area: description;
    max-width: 615px;

    $line-height: 1.4rem;
    white-space: pre-wrap;
    overflow-wrap: break-word;
    
    p {
      line-height: $line-height;
    }

    .collapsed {
      // Limit to three lines
      max-height: 3 * $line-height;
      overflow-y: hidden;
    }

    .expanded {
      max-height: none;
    }

    button {
      margin: 8px 0;
    }
  }
}
