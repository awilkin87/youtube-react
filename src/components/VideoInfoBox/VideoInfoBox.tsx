import { useState } from 'react';
import { Button, Image } from 'semantic-ui-react';
import styles from './VideoInfoBox.module.scss';

export interface VideoInfoBoxProps {
  // TODO
}

const VideoInfoBox: React.FunctionComponent<Readonly<VideoInfoBoxProps>> = function (props) {
  const [collapsed, setCollapsed] = useState(true);

  return (
    <div className={styles.container}>
      <Image className={styles.channelImage} src='http://via.placeholder.com/48x48' circular />
      <div className={styles.videoInfo}>
        <div className={styles.channelName}>Channel Name</div>
        <div className={styles.subscriberCount}>80K Subscribers</div>
      </div>
      <Button className={styles.subscribeButton} color='red'>Subscribe</Button>
      <div className={styles.description}>
        <div className={collapsed ? styles.collapsed : styles.expanded}>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec id tempor massa. Duis in
            suscipit urna. Nunc vel rhoncus lectus. Fusce facilisis mauris at libero mattis porta.
            Donec elementum orci ac consectetur venenatis. In scelerisque, mauris efficitur volutpat
            pellentesque, odio mi congue nisl, sit amet accumsan ipsum nibh at odio.
        </p>
        </div>
        <Button compact onClick={() => setCollapsed(!collapsed)}>
          Show {collapsed ? 'More' : 'Less'}
        </Button>
      </div>
    </div>
  );
}

export default VideoInfoBox;