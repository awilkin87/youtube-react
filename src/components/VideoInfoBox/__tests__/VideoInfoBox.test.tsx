import { shallow } from "enzyme";
import VideoInfoBox from '../VideoInfoBox';

describe('VideoPreview', () => {
  test('Renders collapsed VideoInfoBox', () => {
    const wrapper = shallow(<VideoInfoBox />);
    expect(wrapper).toMatchSnapshot();
  });
});
