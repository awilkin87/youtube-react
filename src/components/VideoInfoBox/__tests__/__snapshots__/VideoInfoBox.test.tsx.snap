// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPreview Renders collapsed VideoInfoBox 1`] = `
<div
  className="container"
>
  <Image
    as="img"
    circular={true}
    className="channelImage"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="videoInfo"
  >
    <div
      className="channelName"
    >
      Channel Name
    </div>
    <div
      className="subscriberCount"
    >
      80K Subscribers
    </div>
  </div>
  <Button
    as="button"
    className="subscribeButton"
    color="red"
  >
    Subscribe
  </Button>
  <div
    className="description"
  >
    <div
      className="collapsed"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec id tempor massa. Duis in suscipit urna. Nunc vel rhoncus lectus. Fusce facilisis mauris at libero mattis porta. Donec elementum orci ac consectetur venenatis. In scelerisque, mauris efficitur volutpat pellentesque, odio mi congue nisl, sit amet accumsan ipsum nibh at odio.
      </p>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show 
      More
    </Button>
  </div>
</div>
`;
